src/AppRouter.tsx

```diff
@@ -51,9 +51,7 @@ export const AppRouter = () => {
             <div>
               <Link to='/expense/new'>Add new expense</Link>
             </div>
-            <div>
-              <BottomNavigation />
-            </div>
+            <BottomNavigation />
```

Comment:
I'd suggest to add BottomNavigation to the router like this: 
```
import Stack from '@mui/material/Stack';
import { styled } from '@mui/material/styles';

const StyledPageContentDiv = styled('div')({
  overflow: 'auto',
  maxHeight: 'calc(100vh - 70px)',
});

const StyledBottomNavigationContainer = styled('div')({
  position: 'absolute',
  bottom: 0,
  left: 0,
  right: 0,
});

...

<Route
        path='/'
        element={
          <Stack
            justifyContent='space-between'
            sx={{ minHeight: '100vh', position: 'relative' }}
          >
            <StyledPageContentDiv>
              <div>
                <Link to='/onboarding/introduction'>
                  Onboarding introduction
                </Link>
              </div>
              <div>
                <Link to='/expense/new'>Add new expense</Link>
              </div>
            </StyledPageContentDiv>
            <StyledBottomNavigationContainer>
              <BottomNavigation />
            </StyledBottomNavigationContainer>
          </Stack>
        }
      />
```

This way the bottom navigation will be attached to the bottom of the page at all times and if the content of the page is too large, the scroll bar will appear.
I added all styles for keeping bottom navigation at the bottom directly to the AppRouter file as 3 different elements need to be styled for it and they depend on each outer. Keeping the code in one place makes it easier to maintain.
Ideally, it would be better to extract the Stack component and everything inside it in a separate file to keep AppRouter clean but it can be done later.